app/forms/mappings/Constraints.scala

```diff
@@ -21,24 +21,38 @@ import utils.Utils.emptyString
 
 trait Constraints {
 
+  private lazy val eoriRegex: String    = "^[A-Z]{2}[0-9A-Z]{1,15}$"
   private lazy val gbnEoriRegex: String = "GBN\\d{11}"
-  private lazy val eoriRegex: String    = "GB\\d{12}"
+  private lazy val gbEoriRegex: String  = "GB\\d{12}"
   lazy val xiEoriRegex: String          = "XI\\d{12}"
   lazy val danRegex: String             = "^[0-9]{7}"
   lazy val canRegex: String             = "^[0-9]{11}"
   lazy val ganRegex: String             = "^[a-zA-Z0-9]{8,10}"
 
   // scalastyle:off cyclomatic.complexity
-  protected def checkEORI(invalidFormatErrorKey: String): Constraint[String] =
-    Constraint {
-      case str if stripWhitespace(str).matches(gbnEoriRegex) => Valid
-      case str if stripWhitespace(str).matches(eoriRegex)    => Valid
-      case str if stripWhitespace(str).matches(danRegex)     => Valid
-      case str if stripWhitespace(str).matches(canRegex)     => Valid
-      case str if stripWhitespace(str).matches(ganRegex)     => Valid
-      case str if stripWhitespace(str).matches(xiEoriRegex)  => Valid
-      case _                                                 => Invalid(invalidFormatErrorKey, eoriRegex)
+  protected def checkEORI(invalidFormatErrorKey: String, isEUEoriEnabled: Boolean = false): Constraint[String] =
+    if (isEUEoriEnabled) {
+      Constraint {
+        case str if isEoriOrAccountNumberValid(str) => Valid
+        case _                                      => Invalid(invalidFormatErrorKey, eoriRegex)
+      }
+    } else {
+      Constraint {
+        case str if stripWhitespace(str).matches(gbnEoriRegex) => Valid
+        case str if stripWhitespace(str).matches(gbEoriRegex)  => Valid
+        case str if stripWhitespace(str).matches(danRegex)     => Valid
+        case str if stripWhitespace(str).matches(canRegex)     => Valid
+        case str if stripWhitespace(str).matches(ganRegex)     => Valid
+        case str if stripWhitespace(str).matches(xiEoriRegex)  => Valid
+        case _                                                 => Invalid(invalidFormatErrorKey, gbEoriRegex)
+      }
     }
 
   protected def stripWhitespace(str: String): String = str.replaceAll("\\s", emptyString).toUpperCase
+
+  private def isEoriOrAccountNumberValid(inputString: String) =
+    stripWhitespace(inputString).matches(danRegex) ||
```

Comment:
Do we care about checking for gb, gbn or xi eori regex in the route that EUEori is enabled?

Reply:
I do not get you question. 

If you mean, that service route should have a check whether a provided Eori is EU eori or not, no. All the checks for EU Eori would be done based on the eu-eori-enabled flag and flow would be decided once you are in a controller and then relevant checks would be done.

If you mean that do we need to check the regex for gb, gbn or XI eori regex in case of EU Eori enabled, the answer is no, we just need to check against EU Eori regex in case of eu-eori-enabled flag is true.

Let me know if this answers your query